inc/Engine/Preload/Fonts.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace WP_Rocket\Engine\Preload;
+
+use WP_Rocket\Admin\Options_Data;
+use WP_Rocket\Event_Management\Subscriber_Interface;
+
+/**
+ * Fonts preload.
+ *
+ * @since 3.6
+ */
+class Fonts implements Subscriber_Interface {
+
+	/**
+	 * WP Rocket Options instance.
+	 *
+	 * @since 3.6
+	 *
+	 * @var Options_Data
+	 */
+	private $options;
+
+	/**
+	 * Return an array of events that this subscriber wants to listen to.
+	 *
+	 * @since  3.6
+	 *
+	 * @param Options_Data $options WP Rocket Options instance.
+	 */
+	public function __construct( Options_Data $options ) {
+		$this->options = $options;
+	}
+
+	/**
+	 * Return an array of events that this subscriber wants to listen to.
+	 *
+	 * @since  3.6
+	 *
+	 * @return array
+	 */
+	public static function get_subscribed_events() {
+		return [
+			'wp_head' => [ 'preload_fonts' ],
+		];
+	}
+
+	/**
+	 * Add the required <link/> tags used to preload fonts.
+	 *
+	 * @since 3.6
+	 */
+	public function preload_fonts() {
+		$fonts = $this->options->get( 'preload_fonts', [] );
+		$fonts = array_map( [ $this, 'sanitize_font' ], $fonts );
+		$fonts = array_filter( $fonts );
+
+		if ( empty( $fonts ) ) {
+			return;
+		}
+
+		$base_url = get_rocket_parse_url( site_url() );
+		$base_url = "{$base_url['scheme']}://{$base_url['host']}";
+
+		foreach ( array_unique( $fonts ) as $font ) {
+			printf(
+				"\n<link rel=\"preload\" as=\"font\" href=\"%s\" crossorigin>",
+				esc_url( get_rocket_cdn_url( $base_url . $font ) )
```

Comment:
I remember we talked about it before, but I don't remember in what context: what's our stance on using `get_rocket_cdn_url()` vs injecting the CDN class in the constructor and using the `rewrite_url()` method?

Reply:
I'd vote for injecting an instance of the CDN into the constructor and then using its `rewrite_url()` method. 

Why? More performant.

- `get_rocket_cdn_url()` gets the container instance, then the CDN instance from the container, and then fires the `rewrite_url()` method. 
- By directly injecting the instance of the CDN, we skip getting the container and CDN instance over and over again as the `foreach` iterates the fonts.
- Benefits: less memory and execution processing time.